### PR TITLE
Feature/hdw/renamecontract

### PR DESCRIPTION
**Pull request by ian-huu**

### CODE REVIEW

**Comment by ian-huu** [comment]
替换命令：
git grep -l '[^.]iost.' | grep -Ev "libv8|vendor"|xargs sed -i "" -E -e '/http/! s/([^.])iost\.([^-". \/@]+)/\1\2.iost/g'

File: rpc/apis.go [hunk 1]
```diff
@@ -331,7 +331,7 @@ func (s *GRPCServer) GetAccountInfo(ctx context.Context, key *GetAccountReq) (*G
 		s.forkDB.Checkout(string(s.bc.LinkedRoot().Block.HeadHash())) // confirm
 	}
 
-	accStr := database.MustUnmarshal(s.visitor.MGet("iost.auth-account", key.ID))
+	accStr := database.MustUnmarshal(s.visitor.MGet("auth.iost-account", key.ID))
```
Comment (wangyu0air) [inline rpc/apis.go]: fix it

File: test/integration/verifier_test.go [hunk 2]
```diff
@@ -402,22 +402,22 @@ func TestAuthority(t *testing.T) {
...
 		So(r.Status.Message, ShouldEqual, "")
-		So(s.Visitor.MGet("iost.auth-account", "myid"), ShouldStartWith, `s{"id":"myid",`)
+		So(s.Visitor.MGet("auth.iost-account", "myid"), ShouldStartWith, `s{"id":"myid",`)
 
-		r, err = s.Call("iost.auth", "AddPermission", array2json([]interface{}{"myid", "perm1", 1}), kp.ID, kp)
+		r, err = s.Call("auth.iost", "AddPermission", array2json([]interface{}{"myid", "perm1", 1}), kp.ID, kp)
 		So(err, ShouldBeNil)
 		So(r.Status.Message, ShouldEqual, "")
-		So(s.Visitor.MGet("iost.auth-account", "myid"), ShouldContainSubstring, `"perm1":{"name":"perm1","groups":[],"items":[],"threshold":1}`)
+		So(s.Visitor.MGet("auth.iost-account", "myid"), ShouldContainSubstring, `"perm1":{"name":"perm1","groups":[],"items":[],"threshold":1}`)
```
Comment (wangyu0air) [inline test/integration/verifier_test.go]: fix it

File: test/integration/vote_producer_test.go [hunk 3]
```diff
@@ -48,18 +48,18 @@ func Test_InitProducer(t *testing.T) {
 		prepareToken(t, s, kp)
 		prepareProducerVote(t, s, kp)
 
-		So(s.Visitor.Get("iost.vote-current_id"), ShouldEqual, `s"1"`)
-		So(s.Visitor.Get("iost.vote_producer-voteId"), ShouldEqual, `s"1"`)
+		So(s.Visitor.Get("vote.iost-current_id"), ShouldEqual, `s"1"`)
```
Comment (wangyu0air) [inline test/integration/vote_producer_test.go]: fix it

File: test/integration/vote_producer_test.go [hunk 4]
```diff
@@ -48,18 +48,18 @@ func Test_InitProducer(t *testing.T) {
...
+		So(s.Visitor.Get("vote_producer.iost-voteId"), ShouldEqual, `s"1"`)
 		Convey("test init producer", func() {
 			for i := 0; i < 12; i += 2 {
-				s.Call("iost.vote_producer", "InitProducer", fmt.Sprintf(`["%v", "%v"]`, testID[i], testID[i]), kp.ID, kp)
+				s.Call("vote_producer.iost", "InitProducer", fmt.Sprintf(`["%v", "%v"]`, testID[i], testID[i]), kp.ID, kp)
 			}
-			So(s.Visitor.Get("iost.vote_producer-pendingProducerList"), ShouldEqual,
+			So(s.Visitor.Get("vote_producer.iost-pendingProducerList"), ShouldEqual,
```
Comment (wangyu0air) [inline test/integration/vote_producer_test.go]: fix it

File: test/integration/vote_producer_test.go [hunk 5]
```diff
@@ -48,18 +48,18 @@ func Test_InitProducer(t *testing.T) {
...
 				`s["IOST4wQ6HPkSrtDRYi2TGkyMJZAB3em26fx79qR3UJC7fcxpL87wTn","IOST54ETA3q5eC8jAoEpfRAToiuc6Fjs5oqEahzghWkmEYs9S9CMKd"`+
 					`,"IOST558jUpQvBD7F3WTKpnDAWg6HwKrfFiZ7AqhPFf4QSrmjdmBGeY","IOST7GmPn8xC1RESMRS6a62RmBcCdwKbKvk2ZpxZpcXdUPoJdapnnh"`+
 					`,"IOST7ZGQL4k85v4wAxWngmow7JcX4QFQ4mtLNjgvRrEnEuCkGSBEHN","IOST7ZNDWeh8pHytAZdpgvp7vMpjZSSe5mUUKxDm6AXPsbdgDMAYhs"]`)
 
-			So(s.Visitor.MKeys("iost.vote-v-1"), ShouldResemble, []string{testID[0], testID[2], testID[4], testID[6], testID[8], testID[10]})
+			So(s.Visitor.MKeys("vote.iost-v-1"), ShouldResemble, []string{testID[0], testID[2], testID[4], testID[6], testID[8], testID[10]})
```
Comment (wangyu0air) [inline test/integration/vote_producer_test.go]: fix it

File: test/integration/vote_producer_test.go [hunk 6]
```diff
@@ -159,55 +159,55 @@ func Test_Vote1(t *testing.T) {
...
 			So(err, ShouldBeNil)
 			So(r.Status.Code, ShouldEqual, tx.Success)
-			r, err = s.Call("iost.vote_producer", "LogInProducer", fmt.Sprintf(`["%v"]`, kp8.ID), kp8.ID, kp8)
+			r, err = s.Call("vote_producer.iost", "LogInProducer", fmt.Sprintf(`["%v"]`, kp8.ID), kp8.ID, kp8)
 			So(err, ShouldBeNil)
 			So(r.Status.Code, ShouldEqual, tx.Success)
-			So(s.Visitor.MGet("iost.vote-v-1", kp6.ID), ShouldEqual, `s["0",false,-1]`)
-			So(s.Visitor.MGet("iost.vote-v-1", kp7.ID), ShouldEqual, `s["0",false,-1]`)
-			So(s.Visitor.MGet("iost.vote-v-1", kp8.ID), ShouldEqual, `s["0",false,-1]`)
+			So(s.Visitor.MGet("vote.iost-v-1", kp6.ID), ShouldEqual, `s["0",false,-1]`)
```
Comment (wangyu0air) [inline test/integration/vote_producer_test.go]: . too many in this file

File: test/integration/vote_test.go [hunk 7]
```diff
@@ -77,9 +77,9 @@ func Test_NewVote(t *testing.T) {
 			So(err, ShouldBeNil)
 			So(r.Status.Code, ShouldEqual, tx.Success)
 			So(s.Visitor.TokenBalance("iost", testID[0]), ShouldEqual, int64(1999999000*1e8))
-			So(s.Visitor.Get("iost.vote-current_id"), ShouldEqual, `s"1"`)
-			So(s.Visitor.MGet("iost.vote-voteInfo", "1"), ShouldEqual, `s{"description":"test vote","resultNumber":2,"minVote":10,"anyOption":false,"unvoteInterval":0,"deposit":"1000"}`)
-			So(s.Visitor.MGet("iost.vote-v-1", "option1"), ShouldEqual, `s["0",false,-1]`)
+			So(s.Visitor.Get("vote.iost-current_id"), ShouldEqual, `s"1"`)
```
Comment (wangyu0air) [inline test/integration/vote_test.go]: fix it

File: test/native/gas_test.go [hunk 8]
```diff
@@ -50,7 +50,7 @@ func gasTestInit() (*native.Impl, *host.Host, *contract.Contract, *account.Accou
 	if err != nil {
 		panic(err)
 	}
-	h.DB().MPut("iost.auth-account", testAcc.ID, database.MustMarshal(string(as)))
+	h.DB().MPut("auth.iost-account", testAcc.ID, database.MustMarshal(string(as)))
```
Comment (wangyu0air) [inline test/native/gas_test.go]: fix it

File: test/native/token721_test.go [hunk 9]
```diff
@@ -17,9 +17,9 @@ var test721DataPath = "./test_data/"
...
-	vi.MPut("iost.auth-account", "user0", database.MustMarshal(`{"id":"user0","permissions":{"active":{"name":"active","groups":[],"items":[{"id":"user0","is_key_pair":true,"weight":1}],"threshold":1},"owner":{"name":"owner","groups":[],"items":[{"id":"user0","is_key_pair":true,"weight":1}],"threshold":1}}}`))
-	vi.MPut("iost.auth-account", "user1", database.MustMarshal(`{"id":"user1","permissions":{"active":{"name":"active","groups":[],"items":[{"id":"user1","is_key_pair":true,"weight":1}],"threshold":1},"owner":{"name":"owner","groups":[],"items":[{"id":"user1","is_key_pair":true,"weight":1}],"threshold":1}}}`))
+	vi.MPut("auth.iost-account", "issuer0", database.MustMarshal(`{"id":"issuer0","permissions":{"active":{"name":"active","groups":[],"items":[{"id":"issuer0","is_key_pair":true,"weight":1}],"threshold":1},"owner":{"name":"owner","groups":[],"items":[{"id":"issuer0","is_key_pair":true,"weight":1}],"threshold":1}}}`))
```
Comment (wangyu0air) [inline test/native/token721_test.go]: fix it

File: test/native/token_test.go [hunk 10]
```diff
@@ -16,9 +16,9 @@ import (
...
-	vi.MPut("iost.auth-account", "user0", database.MustMarshal(`{"id":"user0","permissions":{"active":{"name":"active","groups":[],"items":[{"id":"user0","is_key_pair":true,"weight":1}],"threshold":1},"owner":{"name":"owner","groups":[],"items":[{"id":"user0","is_key_pair":true,"weight":1}],"threshold":1}}}`))
-	vi.MPut("iost.auth-account", "user1", database.MustMarshal(`{"id":"user1","permissions":{"active":{"name":"active","groups":[],"items":[{"id":"user1","is_key_pair":true,"weight":1}],"threshold":1},"owner":{"name":"owner","groups":[],"items":[{"id":"user1","is_key_pair":true,"weight":1}],"threshold":1}}}`))
+	vi.MPut("auth.iost-account", "issuer0", database.MustMarshal(`{"id":"issuer0","permissions":{"active":{"name":"active","groups":[],"items":[{"id":"issuer0","is_key_pair":true,"weight":1}],"threshold":1},"owner":{"name":"owner","groups":[],"items":[{"id":"issuer0","is_key_pair":true,"weight":1}],"threshold":1}}}`))
```
Comment (wangyu0air) [inline test/native/token_test.go]: fix it

File: verifier/simulator.go [hunk 11]
```diff
@@ -63,12 +63,12 @@ func (s *Simulator) SetAccount(acc *account.Account) {
 	if err != nil {
 		panic(err)
 	}
-	s.Visitor.MPut("iost.auth-account", acc.ID, database.MustMarshal(string(buf)))
+	s.Visitor.MPut("auth.iost-account", acc.ID, database.MustMarshal(string(buf)))
```
Comment (wangyu0air) [inline verifier/simulator.go]: fix it

File: vm/host/authority.go [hunk 12]
```diff
@@ -53,7 +53,7 @@ func (h *Authority) isContract(id string) bool {
 
 // ReadAuth read auth
 func ReadAuth(vi *database.Visitor, id string) (*account.Account, contract.Cost) {
-	sa := vi.MGet("iost.auth-account", id)
+	sa := vi.MGet("auth.iost-account", id)
```
Comment (wangyu0air) [inline vm/host/authority.go]: fix it

File: vm/host/host.go [hunk 13]
```diff
@@ -224,11 +224,11 @@ func (h *Host) DestroyCode(contractName string) (contract.Cost, error) {
 		return cost, ErrDestroyRefused
 	}
 
-	owner, co := h.GlobalMapGet("iost.system", "contract_owner", oc.ID)
+	owner, co := h.GlobalMapGet("system.iost", "contract_owner", oc.ID)
 	cost.AddAssign(co)
 	h.PayCost(contract.NewCost(int64(-oldL), 0, 0), owner.(string))
 
-	h.db.MDel("iost.system-contract_owner", oc.ID)
+	h.db.MDel("system.iost-contract_owner", oc.ID)
```
Comment (wangyu0air) [inline vm/host/host.go]: fix it

**Pull request merged**